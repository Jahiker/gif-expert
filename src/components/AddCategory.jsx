import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue) {
      // setCategories((categories) => [inputValue, ...categories]);
      onNewCategory( inputValue.trim() );
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
 