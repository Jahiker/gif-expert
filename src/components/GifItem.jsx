import React from 'react'

import PropsType from "prop-types"

export const GifItem = ({ title, url }) => {
  return (
    <div className='card'>
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropsType.string.isRequired,
  url: PropsType.string.isRequired
}
