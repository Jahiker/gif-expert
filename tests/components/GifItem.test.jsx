import { render } from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";

describe("GifItem test", () => {
  test("should match with snapshot", () => {
    const image = {
      title: "Gif title",
      url: "https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=07c8597afzxwa9byycg9v8lxxrempu6e00jrru5v4tiavcqk&rid=giphy.gif&ct=g",
    };

    const { component } = render(<GifItem {...image} />);
    expect( component ).toMatchSnapshot();
  });
});
