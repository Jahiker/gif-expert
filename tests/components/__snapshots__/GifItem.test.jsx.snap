// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test should match with snapshot 1`] = `undefined`;
